### PR TITLE
Add jdbc connector sample

### PR DESCRIPTION
**Pull request by gurbuzali**
depends on https://github.com/hazelcast/hazelcast-jet/pull/951

### CODE REVIEW

**Comment by gurbuzali** [comment]
verify

File: jdbc/src/main/java/JdbcSink.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        jet = Jet.newJetInstance();
+        Jet.newJetInstance();
+
+        IMapJet<Integer, User> map = jet.getMap(MAP_NAME);
+        IntStream.range(0, 100).forEach(i -> map.put(i, new User(i, "name-" + i)));
+    }
+
+    private void createTable() throws SQLException {
+        DeleteDbFiles.execute(DIR, DB, true);
```
Comment (viliam-durina) [inline jdbc/src/main/java/JdbcSink.java]: Would be better to delete afterwards, not before.

File: jdbc/src/main/java/JdbcSink.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import java.sql.Statement;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+/**
+ * Demonstrates dumping a map's values to a table.
+ */
+public class JdbcSink {
+
+    private static final String DIR = "~";
```
Comment (viliam-durina) [inline jdbc/src/main/java/JdbcSink.java]: Would be better to use `Files.createTempDirectory`, not user home directory. But surprisingly to me, `"~"` works on windows and refers to home directory too.

File: jdbc/src/main/java/JdbcSource.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * Demonstrates reading from a database table and populating IMap
+ */
+public class JdbcSource {
+
+    private static final String DIR = "~";
```
Comment (viliam-durina) [inline jdbc/src/main/java/JdbcSource.java]: Again, use temp directory.

File: pom.xml [hunk 4]
```diff
@@ -51,7 +52,7 @@
         <maven.checkstyle.plugin.version>2.17</maven.checkstyle.plugin.version>
         <log4j.version>1.2.12</log4j.version>
         <!-- for Spring Boot, must be set to Hazelcast version shaded by hazelcast-jet -->
-        <hazelcast.version>3.10</hazelcast.version>
+        <hazelcast.version>3.10.3</hazelcast.version>
```
Comment (cangencer) [inline pom.xml]: this line should not be needed anymore

Comment (gurbuzali) [reply pom.xml]: I guess it is still used by docker compose, if not spring

**Pull request merged**